### ISSUE TITLE
Some flake8 plugins break if flake8-type-checking is installed

### ISSUE THREAD

**Issue by jtraub** [snok/flake8-type-checking#139]
For some reason when `flake8-type-checking` is installed other plugins can stop working. Since this is happening with several plugins I decided to report the problem here (but I might be wrong and the problem is incorrect code in other plugins)

Reproduction steps:

Consider `test.py` file
```python
import os

t = os.path.dirname(os.path.realpath(__file__))
```

Here is the terminal excerpt that highlight the problem using `flake8-use-pathlib` as an example

```
/tmp/flake8_plugin_breakage 
> python -m venv venv

/tmp/flake8_plugin_breakage took 2s 
> source venv/bin/activate

/tmp/flake8_plugin_breakage via 🐍 (venv) 
> pip install flake8 flake8-use-pathlib
Collecting flake8
  Using cached flake8-5.0.4-py2.py3-none-any.whl (61 kB)
Collecting flake8-use-pathlib
  Using cached flake8_use_pathlib-0.3.0-py3-none-any.whl (4.9 kB)
Collecting pycodestyle<2.10.0,>=2.9.0
  Using cached pycodestyle-2.9.1-py2.py3-none-any.whl (41 kB)
Collecting mccabe<0.8.0,>=0.7.0
  Using cached mccabe-0.7.0-py2.py3-none-any.whl (7.3 kB)
Collecting pyflakes<2.6.0,>=2.5.0
  Using cached pyflakes-2.5.0-py2.py3-none-any.whl (66 kB)
Installing collected packages: pyflakes, pycodestyle, mccabe, flake8, flake8-use-pathlib
Successfully installed flake8-5.0.4 flake8-use-pathlib-0.3.0 mccabe-0.7.0 pycodestyle-2.9.1 pyflakes-2.5.0

[notice] A new release of pip available: 22.2.2 -> 22.3
[notice] To update, run: pip install --upgrade pip

/tmp/flake8_plugin_breakage via 🐍 (venv) 
> flake8 test.py 
test.py:4:5: PL120 os.path.dirname("foo/bar") should be replaced by bar_path.parent

/tmp/flake8_plugin_breakage via 🐍 (venv) 
> pip install flake8-type-checking
Collecting flake8-type-checking
  Using cached flake8_type_checking-2.1.3-py3-none-any.whl (19 kB)
Requirement already satisfied: flake8 in ./venv/lib/python3.10/site-packages (from flake8-type-checking) (5.0.4)
Collecting classify-imports
  Using cached classify_imports-4.2.0-py2.py3-none-any.whl (5.4 kB)
Requirement already satisfied: mccabe<0.8.0,>=0.7.0 in ./venv/lib/python3.10/site-packages (from flake8->flake8-type-checking) (0.7.0)
Requirement already satisfied: pycodestyle<2.10.0,>=2.9.0 in ./venv/lib/python3.10/site-packages (from flake8->flake8-type-checking) (2.9.1)
Requirement already satisfied: pyflakes<2.6.0,>=2.5.0 in ./venv/lib/python3.10/site-packages (from flake8->flake8-type-checking) (2.5.0)
Installing collected packages: classify-imports, flake8-type-checking
Successfully installed classify-imports-4.2.0 flake8-type-checking-2.1.3

[notice] A new release of pip available: 22.2.2 -> 22.3
[notice] To update, run: pip install --upgrade pip

/tmp/flake8_plugin_breakage via 🐍 (venv) 
> flake8 test.py 

/tmp/flake8_plugin_breakage via 🐍 (venv) 
> pip uninstall flake8-type-checking
Found existing installation: flake8-type-checking 2.1.3
Uninstalling flake8-type-checking-2.1.3:
  Would remove:
    /tmp/flake8_plugin_breakage/venv/lib/python3.10/site-packages/flake8_type_checking-2.1.3.dist-info/*
    /tmp/flake8_plugin_breakage/venv/lib/python3.10/site-packages/flake8_type_checking/*
Proceed (Y/n)? y
  Successfully uninstalled flake8-type-checking-2.1.3

/tmp/flake8_plugin_breakage via 🐍 (venv) 
> flake8 test.py 
test.py:4:5: PL120 os.path.dirname("foo/bar") should be replaced by bar_path.parent
```

Here I am using CPython ver 3.10.7

**Comment by sondrelg** [comment]
So I've been able to boil it down to [this piece](https://github.com/snok/flake8-type-checking/blob/main/flake8_type_checking/checker.py#L714) of the plugin. We basically modify the `ast` objects like this:

```diff
{
  'value': <ast.Name object at 0x7fb58ec7e830>, 
  'attr': 'path', 'ctx': <ast.Load object at 0x7fb58f5bcdf0>, 
  'lineno': 3, 'col_offset': 20, 'end_lineno': 3, 
  'end_col_offset': 27, 
+ '__flake8-type-checking__parent': 'realpath'
}
```

Without getting too much into the details, this is something we sort of need (though maybe there are better ways of keeping track of parent nodes), and I think I got the inspiration for this from another flake8 plugin though I can't remember which.

What I can't understand, is why this would prevent other plugins from behaving correctly :thinking: Even if the entire tree was modified before sent to other plugins, I'd expect the added attribute to just be ignored

**Comment by sondrelg** [comment]
Wait no, [this](https://github.com/snok/flake8-type-checking/blob/main/flake8_type_checking/checker.py#L726) is the offending code. That makes more sense. I think this should be easily fixable.

**Comment by sondrelg** [comment]
#140 fixes this, I believe. For some reason we were modifying the `node.attr` values, and it doesn't seem like we need to, so I removed it :shrug: v2.2.0 will be out in a few minutes - please let me know if that doesn't solve you issue :slightly_smiling_face:

**Comment by jtraub** [comment]
@sondrelg it now correctly works with other plugins. Thanks a lot for prompt resolution

**Comment by sondrelg** [comment]
Thanks for reporting it!
